contracts/InterchainTokenService.sol

```diff
@@ -551,6 +557,9 @@ contract InterchainTokenService is
         uint256 amount,
         bytes calldata metadata
     ) external payable whenNotPaused {
+        if (sourceAddress == address(0)) revert EmptySourceAddress();
+        if (destinationAddress.length == 0) revert EmptyDestinationAddress();
```

Comment:
Why not check inside `_transmitInterchainTransfer` instead? Some code paths aren't being covered

Reply:
It was because `transmitInterchainTransfer` checks for `EmptySourceAddress()` and `EmptyDestinationAddress()`, which are not required in `_transmitInterchainTransfer`. I thought this might optimize the gas cost since checking the source address would not be necessary. But after removing the `EmptySourceAddress()` check from `transmitInterchainTransfer`, we can move the checkin inside `_transmitInterchainTransfer`. updated.